### PR TITLE
ZEN-28269 Delete subNodes together with a parentNode

### PR DESCRIPTION
**Pull request by karolzen**
https://jira.zenoss.com/browse/ZEN-28269

### CODE REVIEW

File: Products/ZenUI3/browser/resources/js/zenoss/Reports.js [hunk 1]
```diff
@@ -158,12 +158,19 @@ Ext.define('Zenoss.ReportTreePanel', {
...
             fn: function(buttonid) {
                 if (buttonid==='ok') {
+                    report_tree._deleteChilds();
                     report_tree._deleteSelectedNode();
                 }
             },
             buttons: Ext.MessageBox.OKCANCEL
         });
     },
+    _deleteChilds: function () {
```
Comment (jpeacock-zenoss) [inline Products/ZenUI3/browser/resources/js/zenoss/Reports.js]: Please name this function `_deleteChildren` (or `_deleteSubNodes`), i.e. 'children' is the plural form of 'child'.

File: Products/ZenUI3/browser/resources/js/zenoss/Reports.js [hunk 2]
```diff
@@ -158,12 +158,19 @@ Ext.define('Zenoss.ReportTreePanel', {
...
+        var node = this.getSelectionModel().getSelectedNode();
+        while(node.firstChild) {
+            node.removeChild(node.firstChild);
+        }
```
Comment (jpeacock-zenoss) [inline Products/ZenUI3/browser/resources/js/zenoss/Reports.js]: Will this work if there are suborganizers in the organizer you're attempting to delete?  I think a recursive algorithm may be needed here.

File: Products/ZenUI3/browser/resources/js/zenoss/Reports.js (hunk 1 above)
Comment (karolzen) [inline Products/ZenUI3/browser/resources/js/zenoss/Reports.js]: :+1:

File: Products/ZenUI3/browser/resources/js/zenoss/Reports.js (hunk 2 above)
Comment (karolzen) [inline Products/ZenUI3/browser/resources/js/zenoss/Reports.js]: Suborganizers will be deleted as well. This function is only for removing JS's objects.

**Pull request merged**